src/LanguageServer/ProtocolUnitTests/RelatedDocuments/RelatedDocumentsTests.cs

```diff
@@ -0,0 +1,137 @@
+            },
+            CancellationToken.None).ConfigureAwait(false);
+
+        if (useProgress)
+        {
+            Assert.Null(spans);
+            spans = progress!.Value.GetFlattenedValues();
+        }
+
+        AssertEx.NotNull(spans);
+        return spans;
+    }
+
+    [Theory, CombinatorialData]
+    public async Task ReferenceNoDocuments(bool mutatingLspWorkspace, bool useProgress)
+    {
+        var markup1 = """
+            class X
+            {
+            }
+            """;
+
+        var markup2 = """
+            class Y
+            {
+            }
+            """;
+
+        await using var testLspServer = await CreateTestLspServerAsync([markup1, markup2], mutatingLspWorkspace);
+
+        var project = testLspServer.TestWorkspace.CurrentSolution.Projects.Single();
+        var results = await RunGetRelatedDocumentsAsync(
+            testLspServer,
+            project.Documents.First().GetURI(),
+            useProgress: useProgress);
+
+        Assert.Equal(0, results.Length);
+    }
+
+    [Theory, CombinatorialData]
+    public async Task ReferenceSingleOtherDocument(bool mutatingLspWorkspace, bool useProgress)
+    {
+        var markup1 = """
+            class X
+            {
+                Y y;
+            }
+            """;
+
+        var markup2 = """
+            class Y
+            {
+            }
+            """;
+
+        await using var testLspServer = await CreateTestLspServerAsync([markup1, markup2], mutatingLspWorkspace);
+
+        var project = testLspServer.TestWorkspace.CurrentSolution.Projects.Single();
+        var results = await RunGetRelatedDocumentsAsync(
+            testLspServer,
+            project.Documents.First().GetURI(),
+            useProgress: useProgress);
+
+        Assert.Equal(1, results.Length);
+        Assert.Equal(project.Documents.Last().FilePath, results[0].FilePaths.Single());
+    }
+
+    [Theory, CombinatorialData]
+    public async Task ReferenceMultipleOtherDocument(bool mutatingLspWorkspace, bool useProgress)
+    {
+        var markup1 = """
+            class X
+            {
+                Y y;
+                Z z;
+            }
+            """;
+
+        var markup2 = """
+            class Y
+            {
+            }
+            """;
+
+        var markup3 = """
+            class Z
+            {
+            }
+            """;
+
+        await using var testLspServer = await CreateTestLspServerAsync([markup1, markup2, markup3], mutatingLspWorkspace);
+
+        var project = testLspServer.TestWorkspace.CurrentSolution.Projects.Single();
+        var results = await RunGetRelatedDocumentsAsync(
+            testLspServer,
+            project.Documents.First().GetURI(),
+            useProgress: useProgress);
+
+        Assert.Equal(1, results.Length);
+        Assert.Equal(2, results[0].FilePaths!.Length);
+        AssertEx.SetEqual([.. project.Documents.Skip(1).Select(d => d.FilePath)], results[0].FilePaths);
+    }
```

Comment:
Could use additional test for multiple requests and verify you get the changed / no change responses

Reply:
sure.